core/include/detray/tracks/bound_track_parameters.hpp

```diff
@@ -151,6 +151,32 @@ struct bound_track_parameters {
         matrix_operator().element(m_vector, e_bound_qoverp, 0) = qop;
     }
 
+    DETRAY_HOST_DEVICE
+    void flip() {
```

Comment:
I am not quite clear on why we need this. Wouldn't it work to set the step-size to negative values as long as we are in backwards propagation direction? Otherwise, we also have a problem in the handling of overstepping and that needs to be fixed then

Reply:
Flip of bound parameter is not used in navigation. 
I have a question: How does the track find surface candidates unless it is flipped?

Reply:
Good point, we might just need to flip the overstepping check around. The navigator also finds surfaces behind the track position and gives a negative step-size to the stepper, which then automatically reverts the direction of the integration. This is how the overstepping works.

Reply:
OK it makes sense but I am afraid we also need to touch the `contrained_step` setup to make it work :/

Reply:
Hm, I thought I had taken care of that, but let me have a look. If this works, however, I would expect this to be faster than calling a track momentum flip function.

Reply:
It reverts the propagation by the mechanism I described, instead of flipping the track momentum. The step constraints should actually account for this, but I have to check. If this mechanism is somehow flawed, because I overlooked something, then we have to fix the overstepping

Reply:
I did not mean that this PR would break the overstepping. I think it should work fine

Reply:
Ok, from just looking at the code, the step constraints should be fine. They save the constraints as absolute values and multiply the propagation direction back on when they are returned.

Reply:
There actually already is a backwards propagation test for the ```rk_stepper```, but it does not use the overstepping tolerance nor correctly handles overstepping....

Reply:
OK let me check if things work correctly

Reply:
Then flipping the `overstep_tolerance` of track and the `step_size` of stepper would be enough?